### PR TITLE
Preserve ranges in mirroring

### LINKED ISSUE #23855: Mirroring ranged versions removes ranges for most browsers
### What type of issue is this?

Infrastructure issue

### What is the issue?

Both Chrome and Firefox use ranges in this feature:

https://github.com/mdn/browser-compat-data/blob/9832a6516bfe3d8b1d18e503645d0332267c43a0/html/elements/iframe.json#L767-L800

The data we publish (after mirroring) is this:

```
          "allow-forms": {
            "__compat": {
              "description": "<code>sandbox=\"allow-forms\"</code>",
              "source_file": "html/elements/iframe.json",
              "spec_url": "https://html.spec.whatwg.org/multipage/browsers.html#attr-iframe-sandbox-allow-forms",
              "status": {
                "deprecated": false,
                "experimental": false,
                "standard_track": true
              },
              "support": {
                "chrome": {
                  "version_added": "≤49"
                },
                "chrome_android": {
                  "version_added": "49"
                },
                "edge": {
                  "version_added": "79"
                },
                "firefox": {
                  "version_added": "≤49"
                },
                "firefox_android": {
                  "version_added": "49"
                },
                "ie": {
                  "version_added": false
                },
                "oculus": {
                  "version_added": "5.0"
                },
                "opera": {
                  "version_added": "36"
                },
                "opera_android": {
                  "version_added": "36"
                },
                "safari": {
                  "version_added": "≤10.1"
                },
                "safari_ios": {
                  "version_added": "≤10.3"
                },
                "samsunginternet_android": {
                  "version_added": "5.0"
                },
                "webview_android": {
                  "version_added": "49"
                }
              }
            }
          },
```

Note that there are no ranges for any mirrored data except safari_ios, which is done correctly.

### What behavior were you expecting?

Ranges should be preserved for browsers that mirror from Chrome and Firefox.

### What version(s) of BCD is the issue present in?

- [X] The current BCD release
- [X] The current version of the `main` branch

### Do you have anything more you want to share?

cc @queengooborg

### PR DESCRIPTION
**Pull request by foolip**
Fixes https://github.com/mdn/browser-compat-data/issues/23855.

### CODE REVIEW

File: scripts/build/mirror.ts [hunk 1]
```diff
@@ -89,24 +90,24 @@ export const getMatchingBrowserVersion = (
...
     ) {
-      // Handle mirroring for Chromium forks when upstream version is pre-Blink
-      return range ? `≤${r}` : r;
-    } else if (
-      release.engine == sourceRelease.engine &&
-      release.engine_version &&
-      sourceRelease.engine_version &&
-      compare(release.engine_version, sourceRelease.engine_version, '>=')
-    ) {
-      return r;
```
Comment (foolip) [inline scripts/build/mirror.ts:108]: This is the key line. Previously ranges were never propagated here, and now they are except for the first release.

File: scripts/build/mirror.test.ts [hunk 2]
```diff
@@ -176,6 +176,146 @@ describe('mirror', () => {
...
+
+      it('edge 79', () => {
+        const support: InternalSupportBlock = {
+          chrome: {
+            version_added: '≤10',
+          },
+        };
+
+        const mirrored = mirrorSupport('edge', support);
+        // Uncertain early Chrome support doesn't imply possible EdgeHTML
```
Comment (foolip) [inline scripts/build/mirror.test.ts:211]: I was a bit concerned about introducing ranges where there might not be any uncertainty, so I looked at what would be different if Edge 79 doesn't get ranges. Happily, only these 22 paths are affected:

css.properties.filter.svg_elements
html.elements.iframe.sandbox.allow-forms
html.elements.iframe.sandbox.allow-pointer-lock
html.elements.iframe.sandbox.allow-same-origin
html.elements.iframe.sandbox.allow-scripts
html.elements.iframe.sandbox.allow-top-navigation
http.data-url.html_files
http.headers.Content-Security-Policy.form-action.blocks_redirects
http.mixed-content
http.mixed-content.allow_file_urls
http.mixed-content.allow_localhost_url
http.mixed-content.allow_loopback_url
svg.global_attributes.fill-rule
svg.global_attributes.filter
webextensions.api.permissions.Permissions
webextensions.api.permissions
webextensions.api.permissions.contains
webextensions.api.permissions.getAll
webextensions.api.permissions.onAdded
webextensions.api.permissions.onRemoved
webextensions.api.permissions.remove
webextensions.api.permissions.request

Fixing the data in these cases is probably better than adding a rule about transitions from non-WebKit to Blink engines.

**Comment by foolip** [comment]
Tests are failing because this changes when mirroring can be used. Let's make sure we like the new behavior before I update the data to pass the tests.

File: scripts/build/mirror.test.ts (hunk 2 above)
Comment (foolip) [reply scripts/build/mirror.test.ts:211]: Some ranges removed in https://github.com/mdn/browser-compat-data/pull/23858.

Comment (foolip) [reply scripts/build/mirror.test.ts:211]: Another in https://github.com/mdn/browser-compat-data/pull/23861.

Comment (foolip) [reply scripts/build/mirror.test.ts:211]: https://github.com/mdn/browser-compat-data/pull/23865 fixes `filter` for SVG elements. That leaves only the HTTP, SVG and WebExtensions data.

**Review by Elchi3 — APPROVED** [review]
LGTM overall. Thanks for fixing this! Nit to test Firefox Android with more interesting ranges.

The proposed fixes from CI make sense to me. Many can be fixed with `npm run fix` and for the consistency errors, we should remove the ranges from the upstream browser:

- api.RTCStatsReport.type_inbound-rtp.*: chrome should be 79
- css.properties.line-break.*: firefox should be 69

File: scripts/build/mirror.test.ts [hunk 3]
```diff
@@ -176,6 +176,146 @@ describe('mirror', () => {
...
+        const support: InternalSupportBlock = {
+          firefox: {
+            version_added: '≤100',
+          },
+        };
+
+        const mirrored = mirrorSupport('firefox_android', support);
+        assert.deepEqual(mirrored, { version_added: '≤100' });
+      });
+
```
Comment (Elchi3) [inline scripts/build/mirror.test.ts:237]:
```suggestion

      it('firefox_android', () => {
        const support: InternalSupportBlock = {
          firefox: {
            version_added: '≤69',
          },
        };

        const mirrored = mirrorSupport('firefox_android', support);
        assert.deepEqual(mirrored, { version_added: '≤79' });
      });
```

**Comment by lksC695** [comment]
Make one flatform. Using hyper/react j.s/expo  and direct comisions to devolopers  to my pocket.it would collect from house.and can be get out form using smart chip.

**Pull request closed without merging**